File: matrix/nMatrix.c
#include <stdio.h>
#include <stdlib.h>

void createNMatrix(int n, double a[n][n])
{
    int i, j;
    for (i = 0; i < n; i++)
    {

        for (j = 0; j < n; j++)

        {

            a[i][j] = rand() % 20;
        }
    }
    /*

    printf("matrix of n * n order is\n");

    for (i = 0; i < n; i++)

    {

        for (j = 0; j < n; j++)

        {

            printf("%d ", a[i][j]);
        }
        printf("\n");
    }*/
}

void print_mat(int r, int c, int m[r][c])
{
    for (int i = 0; i < r; i++)
    {
        for (int j = 0; j < c; j++)
        {
            printf("\t%d", m[i][j]);
        }
        printf("\n");
    }
    return;
}

void matrixMultiplication(int n, double a[n][n], double b[n][n], double c[n][n])
{
    int i, j, k;
    for (i = 0; i < n; i++)

    {

        for (j = 0; j < n; j++)

        {

            c[i][j] = 0;

            for (k = 0; k < n; k++)

            {

                c[i][j] += a[i][k] * b[k][j];
            }

            //printf("%d ", c[i][j]);
        }

        //printf("\n");
    }
}

int main(int argc, char *argv[])
{
    double a[1000][1000], b[1000][1000], c[1000][1000];
    int n;

    //printf("enter the order of matrix: ");

    n = atoi(argv[1]);

    //printf("enter elements for the first matrix: \n");
    createNMatrix(n, a);

    //printf("enter elements for the second matrix: \n");
    createNMatrix(n, b);

    //printf("this is the result of the multiplication: \n");
    matrixMultiplication(n, a, b, c);

    return 0;
}

File: matrix/nMatrix.py
def createNMatrix(n):
    mat=[]
    for i in range(n): 
        row=[]                              #temporary list to store the row
        for j in range(n): 
            row.append(int(input()))         #add the input to row list
        mat.append(row)                      #add the row to the list
    #print(mat)
    return mat

def print_matrix(n,mat):
    for i in range(n):
        for j in range(n):
            print(mat[i][j], end=" ")
        print()
    return 0

def matrixMultiplication(n,a,b):
    c=[]
    for i in range(n):
        row=[]
        for j in range(n):
            temp = 0
            for k in range(n):
                temp += a[i][k] * b[k][j];
            row.append(temp)
        c.append(row)
    return c


def nmatrix():
    A=[]
    B=[]
    C=[]

    n=int(input("enter the order of matrix: "))

    print("enter elements for the first matrix:")
    A=createNMatrix(n)
    print_matrix(n,A)

    print("enter elements for the second matrix:")
    B=createNMatrix(n)
    print_matrix(n,B)

    print("this is the result of the multiplication:")
    C=matrixMultiplication(n,A,B)
    print_matrix(n,C)

def main():
    nmatrix()

    return 0

if __name__ == '__main__':
    main()
    
